vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java

```diff
@@ -202,13 +202,25 @@ private void initBaseElementsAndListeners() {
         getElement().appendChild(templateElement);
         getElement().appendVirtualChild(container);
 
-        getElement().addEventListener("opened-changed", event -> {
-            if (autoAddedToTheUi && !isOpened()) {
-                getElement().removeFromParent();
-                autoAddedToTheUi = false;
-            }
+        getElement().addEventListener("opened-changed",
+                event -> removeAutoAdded());
+
+        addDetachListener(event -> {
+            // If the notification gets detached, it needs to be marked
+            // as closed so it won't auto-open when reattached.
+            setOpened(false);
```

Comment:
When adding a notification to a dialog, this does not actually close the notification in the browser. The detach listener is called, but the opened state is not synced to the client. Probably because the element was removed from the UI, and then it is not synced anymore.

Example:
```java
Dialog dialog = new Dialog();
dialog.setModal(true);

NativeButton openDialog = new NativeButton("Open", e -> {
    dialog.open();
    Notification.show("Hello", 0, Position.BOTTOM_START); // Should add it to the modal dialog
});
add(openDialog);
```

`Notification.show` is just a shortcut here. A more realistic scenario is to create a persistent Notification with a button to close it, and add it to the dialog. After closing the dialog, the notification can not be interacted with anymore. Doesn't seem to matter if the dialog is modal or not.

Reply:
Just to be clear, this is not necessarily related to the current bug / fix, but it was confusing to see the state being changed to closed, and then still have the notification in the browser.

Reply:
This is indeed similar, but still a different issue. Reported here https://github.com/vaadin/web-components/issues/3823